### PR TITLE
Remove TARGET_IE11 flag

### PR DESCRIPTION
**Pull request by snewcomer**
prep for 4.0!

ref https://github.com/emberjs/data/issues/7724

There are some comments about IE11 in the codebase and features we couldn't use due to lack of APIs.  Will follow up after to remove all comments and apply necessary changes.

This PR ultimately aligns our browser targets with ember.js

- https://github.com/emberjs/ember.js/blob/master/config/targets.js

### CODE REVIEW

File: packages/adapter/tests/dummy/config/targets.js [hunk 1]
```diff
@@ -2,12 +2,6 @@
 
 const browsers = ['last 1 Chrome versions', 'last 1 Firefox versions', 'last 1 Safari versions'];
```
Comment (snewcomer) [inline packages/adapter/tests/dummy/config/targets.js:3]: Perhaps we should change this to `last 2`?  Thoughts @igorT ?

Comment (igorT) [reply packages/adapter/tests/dummy/config/targets.js:3]: We should do whatever Ember does

**Review by igorT — APPROVED** [review]

**Comment by igorT** [comment]
@snewcomer looks like lint is failing

File: packages/adapter/tests/dummy/config/targets.js (hunk 1 above)
Comment (snewcomer) [reply packages/adapter/tests/dummy/config/targets.js]:
```
  browsers = ['last 2 Chrome versions', 'last 2 Firefox versions', 'Safari 12', 'last 2 Edge versions'];
```

Good idea.  Here it is and updated in this PR.

**Pull request merged**